q2_vizard/assets/scatterplot_2d/spec.json

```diff
@@ -9,13 +9,32 @@
     "subtitlePadding": 30
   },
   "autosize": {
-    "type": "pad"
-  },
-  "padding": {
-    "left": 10 
+    "type": "fit",
+    "contains": "padding"
   },
+  "padding": 20,
 
   "signals": [
+    {
+      "name": "width",
+      "init": "isFinite(containerSize()[0]) ? containerSize()[0] : 300",
+      "on": [
+        {
+          "update": "isFinite(containerSize()[0]) ? containerSize()[0] : 300",
+          "events": "window:resize"
```

Comment:
I am struggling to understand what this operation is doing (also, I think this is very similar to raincloud so its probably right). 

My understanding is that you are checking if width container size is finite and if its not you are setting it to 300 and then the window is resizing?

Reply:
I see. Thanks for explaining!